### PR TITLE
boards: nordic: nrf7002dk: Remove NS platforms

### LINKED ISSUE #78942: Board nrf7002dk/nrf5340/cpuapp/ns does not build wifi sample
**Describe the bug**
Board fails to build, see https://github.com/zephyrproject-rtos/zephyr/actions/runs/10918300637/job/30303866778?pr=78621 for log

**Expected behavior**
CI to not be broken

**Impact**
Showstopper, CI fails on unrelated changes

### PR DESCRIPTION
**Pull request by krish2718**
NS platforms are not supported due to lacking crypto support (only supported in NCS), so, remove them from upstream.

Fixes #78942

### CODE REVIEW

**Review by PerMac — CHANGES_REQUESTED** [review]
I believe it is wrong to have a platform in tree which is completely untestable. Also, this is a hacky workaround, not a proper use of tags.

**Comment by PerMac** [comment]
"NS platforms are not supported due to lacking crypto support (only supported in NCS), so, exclude them from testing." If they are not supported, than they shouldn't be tree, no? If you say they need NCS, then why they are not placed in NCS?

**Comment by nordicjm** [comment]
I agree, if the boards here do not work, remove them and put them downstream, they can be brought back when they are working.

**Comment by krish2718** [comment]
> Also, this is a hacky workaround, not a proper use of tags.

Yes, as mentioned in the description.

ok, I will move the board variant downstream.

**Review by nordicjm — DISMISSED** [review]

**Review by jukkar — DISMISSED** [review]

**Review by sachinthegreen — DISMISSED** [review]

**Comment by krish2718** [comment]
@PerMac  please review, as this is gating nRF70 PRs want to get this merged ASAP.

**Review by PerMac — DISMISSED** [review]

File: boards/nordic/nrf7002dk/board.cmake [hunk 1]
```diff
@@ -1,15 +1,8 @@
 # Copyright (c) 2024 Nordic Semiconductor ASA
 # SPDX-License-Identifier: Apache-2.0
 
-if(CONFIG_BOARD_NRF7002DK_NRF5340_CPUAPP_NS OR
-   CONFIG_BOARD_NRF7002DK_NRF5340_CPUAPP_NRF7001_NS)
-  set(TFM_PUBLIC_KEY_FORMAT "full")
```
Comment (anangl) [inline boards/nordic/nrf7002dk/board.cmake:6]: Also the following is only related to non-secure targets so it's unneeded upstream: https://github.com/zephyrproject-rtos/zephyr/blob/733225e4a0322575006cca97b0d1d65f0664843c/boards/nordic/nrf7002dk/board.cmake#L9-L11

File: boards/nordic/nrf7002dk/Kconfig.defconfig [hunk 2]
```diff
@@ -4,15 +4,7 @@
 # SPDX-License-Identifier: Apache-2.0
 
 if BOARD_NRF7002DK_NRF5340_CPUAPP || \
-	BOARD_NRF7002DK_NRF5340_CPUAPP_NS || \
```
Comment (anangl) [inline boards/nordic/nrf7002dk/Kconfig.defconfig:7]: Everything in this file is related to TF-M and does not make much sense if non-secure targets are not present. It'd be better to remove this file completely.

File: boards/nordic/nrf7002dk/Kconfig.defconfig
Comment (anangl) [inline boards/nordic/nrf7002dk/Kconfig.defconfig:1]: Everything in this file is related to combined secure/non-secure builds so it does not make much sense to keep it if non-secure targets are not present.

Comment (krish2718) [reply boards/nordic/nrf7002dk/Kconfig.defconfig:1]: Yeah, I left these in intentionally to make it easier to reinstate but I can remove them as well.

**Review by anangl — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Pull request merged**